src/kademlia/node.cpp

```diff
@@ -621,7 +621,7 @@ struct ping_observer : observer
 			char const* nodes = n.string_ptr();
 			char const* end = nodes + n.string_length();
 
-			while (end - nodes >= 26)
+			while (end - nodes >= 20 + detail::address_size(protocol) + 2)
```

Comment:
do you think it would be hard to unit test the ping_observer for v6 nodes? and, is this the only observer that parses the node string btw?